# The storage name of the point cloud.
string name

# The topic on which to capture a point cloud message. 
# If this is empty, then 'cloud' will be used instead.
string topic

# A point cloud to store.
sensor_msgs/PointCloud2 cloud

# If not empty, transforms the cloud to this frame before storing.
string storage_frame_id

# If not empty, transforms the cloud to this frame in the return result.
string result_frame_id

# A flag to determine whether to reply with the cloud.
int32 action

# Will get a message on topic, or store cloud.
int32 STORE=0

# Will get a message on a topic if it is provided, save, and return it;
# otherwise just returns the existing cloud.
int32 GET=1

# Topic and cloud are ignored, just removes cloud from the server.
int32 CLEAR=2

---

# The cloud that was stored (if return_cloud was true)
sensor_msgs/PointCloud2 cloud

# Any text that might be needed...
int32 result

int32 SUCCESS  = 0
int32 NAME_ERROR  = 1
int32 TOPIC_ERROR  = 2 
int32 TF_ERROR = 3
int32 OTHER_ERROR = 4

---
